src/MerchantToken.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "./MerchantFactory.sol";
+import {console} from "../lib/forge-std/src/console.sol"; //TODO: Remove after testing
+
+// Interface for GasliteDrop
+interface IGasliteDrop {
+    function airdropERC20(
+        address _token,
+        address[] calldata _addresses,
+        uint256[] calldata _amounts,
+        uint256 _totalAmount
+    ) external payable;
+}
+
+// Custom errors
+error INVALID_AMOUNT();
+error UNAUTHORIZED();
+error INVALID_ADDRESS();
+
+contract MerchantToken is MerchantFactory {
```

Comment:
🤔 I know this `MerchantToken is MerchantFactory` relationship is borrowed from our previous factories, but it's always thrown me for a loop. Since this is our Nth factory, I think it's worth reassessing the interface between these two contracts.

In my head, I imagine a Factory deploying Tokens, but the way it's written at the moment, these are one and the same and we've just broken up the code somewhat arbitrarily between two files.

Reply:
one specific refactor i'd suggest, we have 3 deploy related functions:

1. MerchantToken.deployToken()
2. MerchantFactory.deploy()
3. MerchantFactory._deploy()

These could be consolidated into a single MerchantFactory.deploy() with helper functions to break it up as necessary